packages/cli-repl/test/e2e-auth.spec.ts

```diff
@@ -849,6 +849,23 @@ describe('Auth e2e', function() {
         )).to.include('user: \'sha1user\'');
         shell.assertNoErrors();
       });
+      it('provides a helpful error message for SCRAM-SHA-1 in FIPS mode', async function() {
+        const connectionString = await testServer.connectionString();
+        shell = TestShell.start({ args: [
+          connectionString,
+          '--tlsFIPSMode',
+          '-u', 'sha1user',
+          '-p', 'sha1pwd',
+          '--authenticationDatabase', dbName,
+          '--authenticationMechanism', 'SCRAM-SHA-1'
+        ] });
+        await shell.waitForExit();
+        try {
+          shell.assertContainsOutput('Auth mechanism SCRAM-SHA-1 is not supported in FIPS mode');
+        } catch {
+          shell.assertContainsOutput('Could not enable FIPS mode.');
```

Comment:
Just double-checking for my understanding, this would be a sort of a fallback for platforms we test where FIPS would not work just so we don't need to explicitly skipping tests on those?

Reply:
Yes, exactly :) We already have `MONGOSH_TEST_E2E_FORCE_FIPS` as a flag for other tests that ensure that FIPS mode can be enabled when that env var is set, so we know we’re not hitting this branch when we’re not supposed to (that flag is currently only enabled for one e2e test, because we only have one distro in CI with out-of-the-box support, but the important thing is that we test it at all).